tests/svd_solvers.cpp

```diff
@@ -28,11 +28,11 @@ TEST_CASE("svd solver test suite")
         bdc_svd svd_decomposition;
         svd_decomposition.compute(m);
 
-        Eigen::MatrixXd m_reconstructed = svd_decomposition.left()
-                                          * svd_decomposition.values().asDiagonal()
-                                          * svd_decomposition.right();
-        Eigen::MatrixXd approximation_error = m - m_reconstructed;
-        std::cout << "approximation error due to svd:\n" << approximation_error.norm() << "\n";
+        // Eigen::MatrixXd m_reconstructed = svd_decomposition.left()
```

Comment:
A thought, does the Eigen SVD assume column major ordering for the matrices?  Maybe the return types from the svd solver need to be `col_matrix` instead?